src/Compilers/CSharp/Portable/CSharpParseOptions.cs

```diff
@@ -228,6 +173,32 @@ public new CSharpParseOptions WithFeatures(IEnumerable<KeyValuePair<string, stri
             }
         }
 
+        internal override void ValidateOptions(ArrayBuilder<Diagnostic> builder)
+        {
+            ValidateOptions(builder, MessageProvider.Instance);
+
+            // Validate LanguageVersion not SpecifiedLanguageVersion, after Latest/Default has been converted:
+            if (!LanguageVersion.IsValid())
+            {
+                builder.Add(Diagnostic.Create(MessageProvider.Instance, (int)ErrorCode.ERR_BadLanguageVersion, LanguageVersion.ToString()));
+            }
+            
+            if (!PreprocessorSymbols.IsDefaultOrEmpty)
+            {
+                foreach (var symbol in PreprocessorSymbols)
+                {
+                    if (symbol == null)
+                    {
+                        builder.Add(Diagnostic.Create(MessageProvider.Instance, (int)ErrorCode.WRN_DefineIdentifierRequired, "null"));
```

Comment:
After discussion, we're ok with the mismatch between command line and API.

---
In reply to: [105762061](https://github.com/dotnet/roslyn/pull/17113#discussion_r105762061) [](ancestors = 105762061,105716480)